assets/src/navLinkData.ts

```diff
@@ -1,19 +1,17 @@
 import React from "react"
 import { fullStoryEvent } from "./helpers/fullStory"
-import {
-  DiamondTurnRightIcon,
-  LadderIcon,
-  MapIcon,
-  SearchMapIcon,
-} from "./helpers/icon"
+import { LadderIcon, MapIcon, SearchMapIcon } from "./helpers/icon"
 import inTestGroup, { TestGroups } from "./userInTestGroup"
+import { DetourNavIcon, NavIconProps } from "./helpers/navIcons"
 
 type HTMLElementProps = React.PropsWithoutRef<React.HTMLAttributes<HTMLElement>>
 
 export interface LinkData {
   title: string
   path: string
-  navIcon: React.JSXElementConstructor<HTMLElementProps>
+  navIcon:
+    | React.JSXElementConstructor<HTMLElementProps>
+    | ((props: NavIconProps) => React.JSX.Element)
```

Comment:
This feels unideal to me, but I'm not sure how to convince typescript that this is okay without doing this.

Reply:
IMO, the best way to handle this would be to change navIcon to a `ReactNode` and stop having the nav supply classes to the content it's trying to render, but the way our CSS is built is probably less friendly to that idea.